src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1139,6 +1139,22 @@ private void ImportCpOpj(int token)
 
         private void ImportUnbox(int token, ILOpcode opCode)
         {
+            TypeDesc type = ResolveTypeToken(token);
+            if (type.IsNullable)
+                throw new NotImplementedException();
+
+            var value = _stack.Pop();
+
+            var boxPtr = LLVM.BuildPointerCast(_builder, value.LLVMValue, LLVM.PointerType(GetLLVMTypeForTypeDesc(type), 0), String.Empty);
+            if (opCode == ILOpcode.unbox)
+            {
+                _stack.Push(new ExpressionEntry(StackValueKind.NativeInt, string.Empty, boxPtr, type.MakePointerType()));
```

Comment:
According to the ECMA spec, this should generate a TypeLoadException if it's not a value type. You don't need to put that in here since we can't do that anyway, but having a check that throws would at least keep us from accidentally compiling something bad here.